File: function.exs
# function.exs
defmodule FunctionExample do
  def my_function(param) do
    x = param + 10
    x = if x > 15 do
      x - 5
    else
      x + 5
    end
    Enum.reduce(0..2, x, fn i, acc -> acc + i end)
  end
end

IO.write("Enter an integer: ")
input = IO.gets("") |> String.trim()
param = String.to_integer(input)
if param == 0 and input != "0" do
  IO.puts("Invalid input")
else
  IO.puts(FunctionExample.my_function(param))
end

File: function.rb
# function.rb
def my_function(param)
  x = param
  x += 10
  if x > 15
      x -= 5
  else
      x += 5
  end
  for i in 0..2
      x += i
  end
  return x
end

print "Enter an integer: "
input = gets.chomp
param = input.to_i

if param == 0 && input != "0"
  puts "Invalid input"
else
  puts my_function(param)
end
